persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -244,69 +258,228 @@ private void authorizeByToken(ByteBuffer token, CQLStatement statement) {
           castStatement.keyspace(),
           castStatement.name());
     } else if (statement instanceof SchemaTransformation) {
-      SchemaTransformation castStatement = (SchemaTransformation) statement;
```

Comment:
On line 246:

```java
        authorization.authorizeSchemaWrite(
            authToken, castStatement.keyspace(), castStatement.name(), Scope.TRUNCATE);
```

Should truncate be considered at data write: `authorizeDataWrite()` like a delete?

Reply:
Good point, fixing